src/sentry/scripts/tsdb/cmsketch.lua

```diff
@@ -318,6 +318,166 @@ function Sketch:increment(items)
+            table.insert(
+                items,
+                {key, tonumber(value)}
+            )
+        end
+        self:increment(items)
+    elseif not table.is_empty(source_estimators) and not table.is_empty(destination_estimators) then
+        -- Merge estimators.
+        for key, value in pairs(source_estimators) do
+            redis.call('HINCRBY', self.estimates, key, value)
+        end
+
+        -- Rebuild the index by using the keys from both indices and the new estimates.
+        local members = {}
+
+        for key, _ in pairs(source_index) do
+            members[key] = true
+        end
+
+        for key, _ in pairs(destination_index) do
+            members[key] = true
+        end
+
+        redis.call('DEL', self.index)
+        for key, _ in pairs(members) do
+            redis.call(
+                'ZADD',
+                self.index,
+                reduce(
+                    math.min,
+                    map(
+                        function (coordinates)
+                            return self:observations(coordinates)
+                        end,
+                        self:coordinates(key)
+                    )
+                ),
+                key
+            )
+        end
+
+        -- Remove extra items from the index.
+        redis.call('ZREMRANGEBYRANK', self.index, 0, -self.configuration.index - 1)
+    elseif table.is_empty(source_estimators) and not table.is_empty(destination_estimators) then
+        -- If we're just writing the index values (and not estimators) to the
+        -- destination, we can just directly increment the sketch which will
+        -- take care of estimator updates and index truncation.
+        local items = {}
+        for key, value in pairs(source_index) do
+            table.insert(
+                items,
+                {key, tonumber(value)}
+            )
+        end
+        self:increment(items)
+    elseif not table.is_empty(source_estimators) and table.is_empty(destination_estimators) then
+        -- Build the estimators for the side that doesn't have them.
+        for key, score in pairs(destination_index) do
+            local coordinates = self:coordinates(key)
+            local estimates = map(
+                function (coordinate)
+                    return self:observations(coordinate)
+                end,
+                coordinates
+            )
+            for _, item in pairs(zip({coordinates, estimates})) do
+                local coordinate, estimate = unpack(item)
+                local update = math.max(score, estimate)
+                if estimate ~= update then
+                    redis.call(
+                        'HSET',
+                        self.estimates,
+                        struct.pack('>HH', unpack(coordinate)),
+                        update
+                    )
+                end
+            end
+        end
+
+        -- Merge the estimators.
```

Comment:
since these next two parts are the same as case 2 can we DRY it into a function or method? or is there a difference i missed?